### PR TITLE
[#2] Chat Interface

### PR DESCRIPTION
**Pull request by dodoels**
- Basic chat interface mostly for POC (lacking interaction)
- Basic VueJS setup, there are more libraries need to be integrated

### CODE REVIEW

File: src/web_app/chatbot/index.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>justiceai</title>
```
Comment (ghost) [inline src/web_app/chatbot/index.html]: proceZeus*

To be changed through the whole PR. JusticeAI is not the name agreed upon by our stakeholder.

File: src/web_app/chatbot/src/components/Hello.vue [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    <button v-on:click="openChatBot">Start</button>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'welcome',
+  data () {
+    return {
+      msg: 'Welcome to JusticeAI! To get started, please tell us what you’d like to do.',
```
Comment (mihaiqc) [inline src/web_app/chatbot/src/components/Hello.vue]: proceZeus*

**Comment by arekmano** [comment]
Do we want to commit to Webpack? I'm not very familiar with it. I know its powerful but it requires a lot of configuration. I defer to F.E. experts for their opinion.

**Comment by dodoels** [comment]
@arekmano  Webpack is generic when you install VueJS. I do not think that using other build tools is a good idea.

**Comment by dodoels** [comment]
Here is other build tool work with VueJS
https://github.com/sohelamin/vue-example

**Comment by lancelafontaine** [comment]
@arekmano this webpack config was mostly generated by `vue-cli` which manages everything you would need. Unless we need to change something drastically, we shouldn't have to ever change these webpack config files. I agree that Webpack is definitely unwieldy and config-heavy, but unfortunately it's the future of everything related to front-end development. On the bright side, everything reasonable that we would want to do is already existing as an npm script.

**Comment by lancelafontaine** [comment]
Since this is all static content,it should probably be in a separate directory with something like `nginx` like `web_client_proxy` or something. At the moment there's a Dockerfile for a web backend in this directory also. Let's sort out directory structure / dockerization of front-end in a future issue (like #4 for example).

File: src/web_app/chatbot/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,9 @@
+root = true
```
Comment (lancelafontaine) [inline src/web_app/chatbot/.editorconfig:1]: Will need to make sure this aligns with our linting strategy

File: src/web_app/chatbot/package.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
+{
+  "name": "justiceai",
```
Comment (lancelafontaine) [inline src/web_app/chatbot/package.json]: proceZeus

File: src/web_app/chatbot/.editorconfig (hunk 3 above)
Comment (lancelafontaine) [reply src/web_app/chatbot/.editorconfig:1]: Actually, just realized this PR also contains our linting. Never mind 👍

File: src/web_app/chatbot/package.json [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  "version": "1.0.0",
+  "description": "Cyberjustice Laboratory Legal Chatbot",
+  "author": "CYberjustice Laboratory",
+  "private": true,
```
Comment (lancelafontaine) [inline src/web_app/chatbot/package.json]: It's public so can remove this line

**Review by lancelafontaine — APPROVED** [review]
All configuration looks good, and running the application, tests and linting works as expected 👍

**Review by ghost — APPROVED** [review]

File: src/web_app/chatbot/.editorconfig (hunk 3 above)
Comment (dodoels) [reply src/web_app/chatbot/.editorconfig:1]: I want to use airbnb linting in next iteration so will be more changes coming

**Pull request merged**